packages/internal/src/generate/watch.ts

```diff
@@ -28,67 +29,190 @@ import {
   mirrorPathForCell,
 } from './typeDefinitions'
 
+// Can't use `isPageFile()` from ../files because that ones looks at the
+// content of the file, and that doesn't work for deleted files that we
+// sometimes have in the watcher callback
+function isPageFile(p: string) {
+  const { name } = path.parse(p)
+
+  // A page must end with "Page.{jsx,js,tsx}".
+  if (!name.endsWith('Page')) {
+    return false
+  }
+
+  // A page should be in the `web/src/pages` directory.
+  if (!isFileInsideFolder(p, getPaths().web.pages)) {
+    return false
+  }
+
+  return true
+}
+
 const rwjsPaths = getPaths()
 
-const watcher = chokidar.watch('**/src/**/*.{ts,js,jsx,tsx}', {
+const action = {
+  add: 'Created',
+  unlink: 'Deleted',
+  unlinkDir: 'Deleted',
+  change: 'Modified',
+}
+
+const apiWatcher = chokidar.watch('src/**/*.{ts,js}', {
   persistent: true,
   ignored: ['node_modules', '.redwood'],
   ignoreInitial: true,
-  cwd: rwjsPaths.base,
+  cwd: rwjsPaths.api.base,
   awaitWriteFinish: true,
 })
 
-const action = {
-  add: 'Created',
-  unlink: 'Deleted',
-  change: 'Modified',
-}
+const dirWatcher = chokidar.watch('src', {
```

Comment:
Was there a reason to separate these watchers? I'm not 100% sure, but I think spinning up more watchers is discouraged, especially on systems with a big FS penalty e.g. Windows, and certain scenarios with docker.

Reply:
Yes. It's a trade-off I guess. This `dirWatcher` is going to be watching a lot more files because of the less restrictive pattern. To limit that, I split it up so that I could keep the more restrictive pattern for the API side. Because on that side we don't have to watch for folder deletions.

Reply:
I can't speak with full confidence which is better really! Let's leave it as is, but worth doing a check on Windows maybe, but its also super hard to quantify the impact of something like this.